.github/workflows/tests_and_cov.yml

```diff
@@ -19,20 +19,20 @@ jobs:
         with:
           python-version: 3.11
 
-      - name: Install deps
-        run: |
-          pip install --upgrade --upgrade-strategy eager -r requirements-dev.txt -r requirements.txt -e .
-          pip check
-
         # Cache package installation step to speed up the following step
       - uses: actions/cache@v4
```

Comment:
Caching done before pip installing will allow to skip another pip install of the repo for each test in the matrix

Reply:
For each test group, right? For each test or test file would be absolutely huge!